code/src/Components/TaskItem.js

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from "react"
+import { useDispatch } from "react-redux"
+
+import { todo } from "../Reducers/todo"
+
+export const TaskItem = ({ item }) => {
+  const dispatch = useDispatch();
+  const [checked, setChecked] = useState(item.isComplete)
+  const handleCheckboxClick = () => {
+    setChecked();
+    dispatch(todo.actions.setComplete(item.id))
+  }
+
+  //Deleting one element from the redux store
+  const onItemDelete = (id) => {
+    dispatch(todo.actions.removeItem({ id: item.id }))
+  }
+
+  return (
+    <section className="task-container">
+      <div className="task-box">
+        <label className="custom-checkbox">
+          <input
+            type="checkbox"
+            checked={checked}
```

Comment:
```suggestion
            checked={item.isCompleted}
```

so then you can remove line 8, no need to having a local state here.